### ISSUE TITLE
Unify email columns to ensure emails are sent

### ISSUE THREAD

**Issue by timolegros** [hicommonwealth/commonwealth#5385]
## Describe the bug

We have 2 email columns: `Profiles.email` and `User.email`. `User.email` is only set if a user signs in with an email and `Profiles.email` is only set if a user updates their profile.

## To Reproduce

Note: Only reproducible locally if you have a local sendgrid API key and you modify the code to ensure no emails are sent to real users.

1. Sign in with a web-wallet.
2. Subscribe to new thread notifications in a community of your choice
3. Enable emails for the new thread subscription from the notification settings page
4. With another browser/account create a thread in the community
5. You won't receive an email even though you signed up for one.

Now:
1. Sign in with your email
2. Subscribe to new thread notifications in a community of your choice
3. Enable emails for the new thread subscription from the notification settings page
4. With another browser/account create a thread in the community
5. You should receive an email.

## Expected behavior

- Email notifications should be sent regardless if a user signed in with an email or updated their profile with the email.
- We should only have a single email column per user (`User.email` or `Profile.email` but not both).

**Comment by CowMuon** [comment]
@Rotorsoft for visibility.

**Comment by Rotorsoft** [comment]
@timolegros couple of questions (assuming our target state):

- Seems like we allow a user (with a primary login email in User) have multiple profiles with different emails. Is this a requirement we will keep?
- When a user enables email notifications, can he select which email from his profiles?
- Do we save the selected email in the notifications table?

**Comment by timolegros** [comment]
> Seems like we allow a user (with a primary login email in User) have multiple profiles with different emails. Is this a requirement we will keep?
I don't think so given https://github.com/hicommonwealth/commonwealth/pull/5336. @mhagel did you confirm with product that we are dropping the multiple profiles per user before your PR?

> When a user enables email notifications, can he select which email from his profiles?
Manual entry only no drop-down/selection.

> Do we save the selected email in the notifications table?
When enabling notification emails from the notification settings, we update `User.email`. No user data is stored in the `Notifications` table. `NotificationsRead` table stores a `user_id` and `notification_id` but no email there either.

**Comment by Rotorsoft** [comment]
Thanks @timolegros. Let's touch base and document what's the final state (aka Model).

**Comment by mhagel** [comment]
>I don't think so given https://github.com/hicommonwealth/commonwealth/pull/5336. @mhagel did you confirm with product that we are dropping the multiple profiles per user requirement before your PR?

Not confirmed with product. Guidance from @jnaviask was that use cases for `is_default` were never implemented.

**Comment by jnaviask** [comment]
- We do not support multiple profiles.
- We do need to confirm with product that we want the user's login email to be the same as their visible contact email -- this may be undesirable, so it may make sense to have two separate emails. We should also confirm which email will receive notifications (likely login email), or whether this should be a third, separate email.

**Comment by Rotorsoft** [comment]
> * We do not support multiple profiles.
> * We do need to confirm with product that we want the user's login email to be the same as their visible contact email -- this may be undesirable, so it may make sense to have two separate emails. We should also confirm which email will receive notifications (likely login email), or whether this should be a third, separate email.

Should we then remove the Profile table (with 1:n relationship to User)? Sounds like User = Profile
Regarding notifications, we can follow GitHub's strategy, they allow to configure specific emails by project/community... flexible and convenient

**Comment by jnaviask** [comment]
Would like product clarification here on how email management should work @zakhap @sachben91

**Comment by CowMuon** [comment]
We can wait until Monday for an answer here @zakhap @sachben91 but also, we don't want this to just get dropped and we have to chase down the answer.

**Comment by dillchen** [comment]
i do not believe we should tackle this until -> https://www.notion.so/buildcommon/PRD-Notifications-Support-for-Mobile-and-Stakes-e15a06b52a7f4d3e91f3a307ca8cd0f1?pvs=4

There is a column in Delivery Mechanisms that we should use

**Comment by timolegros** [comment]
> i do not believe we should tackle this until -> https://www.notion.so/buildcommon/PRD-Notifications-Support-for-Mobile-and-Stakes-e15a06b52a7f4d3e91f3a307ca8cd0f1?pvs=4
> 
> There is a column in Delivery Mechanisms that we should use

@dillchen The delivery mechanisms column doesn't affect this work. This ticket is about ensuring that we have consistent email handling across user accounts and profiles. The email will not be stored in the delivery mechanisms column of the Subscriptions table.

**Comment by dillchen** [comment]
Ok

Sent via Superhuman iOS ***@***.***>
